### PR TITLE
[Feature] CAIN model

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/models/video_interpolater/cain.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        # generator
+        pred = self.generator(inputs, padding_flag=True)
+
+        # loss
+        losses = dict()
+        log_vars = dict()
+        losses['loss_pix'] = self.pixel_loss(pred, target)
+
+        # parse loss
+        loss, log_vars = self.parse_losses(losses)
```
Comment (ckkelvinchan) [inline mmedit/models/video_interpolater/cain.py:49]: In `BasicInterpolater`, `self.parse_losses` is not done in this function. Is there any reason we do not follow `BasicInterpolater`?

File: mmedit/models/video_interpolater/cain.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        Returns:
+            dict: Output results, which contain either key(s)
+                1. 'eval_result'.
+                2. 'inputs', 'pred'.
+                3. 'inputs', 'pred', and 'target'.
+        """
+
+        # generator
+        with torch.no_grad():
+            pred = self.generator(inputs, padding_flag=True)
```
Comment (ckkelvinchan) [inline mmedit/models/video_interpolater/cain.py:93]: So `padding_flag` is always True?

Comment (Yshuo-Li) [reply mmedit/models/video_interpolater/cain.py:93]: No, `padding_flag` is False in training (in the original repo)

File: mmedit/models/video_interpolater/cain.py (hunk 1 above)
Comment (Yshuo-Li) [reply mmedit/models/video_interpolater/cain.py:49]: The subclasses should overwrite the function `forward_train`, `forward_test`, and `train_step`.

Comment (ckkelvinchan) [reply mmedit/models/video_interpolater/cain.py:49]: `train_step` is not modified in this class. It seems that `self.parse_losses` is called twice. 

If the functionalities are identical to the parent class, I think we can directly inherit from it.

File: mmedit/models/video_interpolater/cain.py (hunk 2 above)
Comment (ckkelvinchan) [reply mmedit/models/video_interpolater/cain.py:93]: But it is also set to True in training? (line 41)

File: mmedit/models/video_interpolater/cain.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            dict: Returned output, which includes:
+                log_vars, num_samples, results (inputs, target and pred).
+
+        """
+        # data
+        inputs = data_batch['inputs']
+        target = data_batch['target']
+
+        # generator
+        pred = self.generator(inputs, padding_flag=False)
```
Comment (ckkelvinchan) [inline mmedit/models/video_interpolater/cain.py:41]: @Yshuo-Li I think the only difference between this function and that in `BasicInterpolator` is this line, where an additional argument `padding_flag=False` is required? If yes, I think you can simply copy the whole function to here, and then modify this line. 

Another way is to set a default value in the backbone. If the default value is False, then this function can be inherited from `BasicInterpolator`. If the default is True. Then `forward_test` can be inherited.

**Review by ckkelvinchan — APPROVED** [review]

**Comment by ckkelvinchan** [comment]
ImportError: cannot import name 'PILLOW_VERSION' from 'PIL' (/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/PIL/__init__.py)

It seems that the previous errors occur. Maybe try pulling from master and see if the problem persists

**Review by ckkelvinchan — APPROVED** [review]

**Pull request merged**